### DISCUSSION TITLE
Images folder structure

### DISCUSSION THREAD

**Discussion by ceejayemm** [photoprism/photoprism#2387] [Q&A]
I have recently found PhotoPrism an like it a lot. I have one small issue that I would like help with if anybody be so kind.

My image library has been around for a while and is organised into subject trees eg Holidays, Family, Hobbies etc. Each main folder has one one more subfolders at one or more levels eg

Holidays
     Cornwall
           April 2015
           October 2016
    France
           June 2016
           July 2017
   Italy
           May 2018
Family
      Parents
      Family Occasions
           Wedding
            Silver Wedding
    Jones
           Callum
           Johnny
           Michael

When these folders get imported in PhotoPrism the folder structure gets flattened, mixing up the folders in no particular order.  Is there any way to preserve the original folder structure?  I have tried adding Folders into Albums but that doesn’t help. 

Any other suggestions?

Thanks

Chris

**Comment by graciousgrey** [comment]
If you use indexing instead of importing, your folder structure will be preserved: 

https://docs.photoprism.app/user-guide/library/

Currently only the [originals section](https://docs.photoprism.app/user-guide/library/files/) shows folders hierarchically.

The [folder section](https://docs.photoprism.app/user-guide/organize/folders/) shows all your folders without hierarchy. It is planned to optionally display folders hierarchically:

https://github.com/photoprism/photoprism/issues/1536

Until then you can use the search within the folder view e.g. if you only want to see subfolders of your Holiday folder search for Holiday

**Comment by ceejayemm** [comment]
Theresa

I am getting use to the PhotoPrism terminology, I did indeed Index my
image library.  I have subsequently found the Indexed folder structure
in the Originals area, which is fine as far as it goes.  The 'Folders'
area is, to me, a much more logical for this than Originals but it is
good to hear that this might eventually be mirrored in Folders - this
would definitely get my vote. 

I know this is an evolving application and I am very interested in
where this application might go.

Many thanks

Chris

**Comment by Manfred73** [comment]
Displaying folders hierarchically and selecting any subfolder to add them to an album would be great.
I've gathered many photos and videos over the last 25 years and divided them into 6 main categories (main folders) which each contain many subfolders (which again can contain many subfolders), e.g.

```
|--01_Holidays
|  |
|  |--2012
|  |  |
|  |  |--Austria
|  |  |  |
|  |  |  |--2012-08-08
|  |  |     |
|  |  |     |.............
|  |  |--UK
|  |  |  |
|  |  |  |--2012-12-10
|  |  |     |
|  |  |     |.............
|  |  |--.........
|  |
|  |--2013, etc.
|
|--02_Work
|  |
|  |-- etc.
```

Now every subfolder shows up as a separate tile in the folders view, which makes it hard to find all photos for let's say 01_Holidays\2012\Austria

**Comment by philfry9** [comment]
Deal breaker. Will happily pay for commercial edition when this rolls out if I haven't settled into something else. Can't imagine that this isn't a huge deal for tons of users, pro and non alike.
